supplementary/envelopes_analysis/field_envelopes.py

```diff
@@ -24,26 +24,23 @@
 
 # plotting different envelopes of electric field and intensity, and their spectra
 print("\nEnvelope profiles and spectra calculated")
-colors = plt.cm.viridis(np.linspace(0, 0.9, len(envelope_types)))
+colormap = plt.cm.viridis(np.linspace(0, 0.9, len(envelope_types)))
+colors = dict(zip(envelope_types, colormap))
```

Comment:
This is perhaps a bit too clever :sweat_smile: But it makes the code below nicer since we can index into this dictionary by envelope_type key